core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -831,7 +831,7 @@ public void next() throws IOException {
       try {
         _next();
       } catch (IOException ioe) {
```

Comment:
Reviewing this I was wondering why the RFile object was even reused.  Looking into it I think the code was only not reusing when an IOException happened.  Opened #3623 for that, can do the same here, on io exception or runtime exception close w/ true.

```suggestion
      } catch (IOException |RuntimeException  ioe) {
```

Reply:
Added in d409537